shared/src/types/ibc.rs

```diff
@@ -188,3 +196,420 @@ impl ClientUpgradeData {
             .map_err(|e| Error::DecodingError(e.to_string()))
     }
 }
+
+/// Data to initialize a connection
+#[derive(
+    Debug, Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub struct ConnectionOpenInitData {
+    /// The corresponding client ID
+    client_id: String,
+    /// The corresponding counterparty
+    counterparty: Vec<u8>,
+    /// The version
+    version: Vec<u8>,
+    /// The delay period as (secs, nanos)
+    delay_period: (u64, u32),
```

Comment:
we can make this a little nicer by adding a type for this in `time.rs` module, e.g. `DurationNano { secs: u64, nanos: u32 }` with `impl From<std::time::Duration> for DurationNano` and `impl From<DurationNano> for std::time::Duration`

Reply:
That looks great! Modified